compiler/src/utils/config.re

```diff
@@ -306,8 +306,25 @@ let option_conv = ((prsr, prntr)) => (
     | Some(x) => prntr(ppf, x),
 );
 
-let optimizations_enabled =
-  toggle_flag(~doc="Disable optimizations.", ~names=["O0"], true);
+type optimization_level =
+  | Level_zero
+  | Level_one
+  | Level_two
+  | Level_three;
+
+let optimization_level =
+  opt(
+    ~doc="Set the optimization level.",
+    ~names=["O"],
+    ~conv=
+      Cmdliner.Arg.enum([
+        ("0", Level_zero),
+        ("1", Level_one),
```

Comment:
These are useful for compiler authors, but aren't really useful for end users (since the programs may not work). We could maybe consider removing these, but I'm not sure how we as compiler authors would access these for development.

Reply:
> We could maybe consider removing these, but I'm not sure how we as compiler authors would access these for development.

Sorry, I'm not sure what you mean by this? I'm not sure I understand the issue

Reply:
`-O0` and `-O1` are basically unsupported, so I was saying we could choose to only allow `-O2` and `-O3`. But to your point that it's largely for devs, it's probably fine.

Reply:
I like leaving us room to expand the usage of optimization levels (and thus keeping generic).